ships/0038-release-branching-backports.md

```diff
@@ -127,78 +129,260 @@ such situations.
-- Applying the process "retroactively" to build, cli, and operator repos.
-
-### Test Plan
-
-TBD
-
-**Note:** *Section not required until targeted at a release.*
+#### Release Branching Workflow
+
+Shipwright has set up the `.github` repository which - amongst other features -
+allows GitHub actions to be shared consistently across the organization. We already
+have a standing "add to project" workflow that adds all new issues and pull requests
+to the main "Shipwright Overview" GitHub project. [1]
+
+We can craete a reusable "release branching" workflow [2] as follows:
+
+- Use `workflow_call` as the trigger, to be run on the main branch. [2]
+- Receive a semantic version as input, to then create the release branch name.
+- Create the release branch via standard git commands, ensuring the workflow is
+  granted write permission to the appropriate repository.
+
+Example (`shipwright-io/.github/.github/workflows/release-branch.yml`):
+
+```yaml
+name: Release Brancher
+on:
+  workflow_call:
+    inputs:
+      release-version:
+        required: true
+        type: string
+      git-ref:
+        required: false
+        type: string
+jobs:
+  release-brancher:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.git-ref }}
+    - name: Create release branch
+      env:
+        RELEASE_VERSION: release-${{ inputs.release-version }}
+      run: |
+        git branch ${RELEASE_VERSION}
+        git switch ${RELEASE_VERSION}
+        git push --set-upstream origin ${RELEASE_VERSION}
+```
 
+Each repository can then be onboarded using a starter workflow that is hosted in the
+.github repository (`.github/workflow-templates/release-branch.yml`) [3]:
+
+```yaml
+name: Create Release Branch
+on:
+  workflow_dispatch:
+    inputs:
+      release-version:
+        required: true
+        type: string
+        description: "Semantic version for the release branch (vX.Y format)"
+jobs:
+  create-release-branch:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    uses: shipwright.io/.github/.github/workflows/release-branch.yml@main
+    with:
+      release-version: ${{ inputs.release-version }}
 ```
-Consider the following in developing a test plan for this enhancement:
 
-- Will there be e2e and integration tests, in addition to unit tests?
-- How will it be tested in isolation vs with other components?
+[1] https://github.com/shipwright-io/.github/blob/main/.github/workflows/issues.yml
+
+[2] https://docs.github.com/en/actions/using-workflows/reusing-workflows
+
+[3] https://docs.github.com/en/actions/using-workflows/creating-starter-workflows-for-your-organization
+
+#### Release Workflows
+
+Component repositories will retain the core of their existing release workflows.
+However, each will need to be modified so the release workflow runs against the
+appropriate `release-vX.Y` branch:
+
+- `build`: The `release.yml` workflow will receive a new required parameter,
```

Comment:
Fo build, we will get a PR that updates the readme. This PR we either simply manually fix, or we must extend the logic. I guess by default it would modify the readme from the current branch and open the PR against main. This will probably not be mergable.

Reply:
Looking at the action doc [1], the `base` parameter can be used to specify the PR base branch, and it defaults to "the branch checked out in the workflow." So you could have some temporary skew between the release branch and `main` if leave it as is (or set main to always reference the "latest" release).

[1] https://github.com/marketplace/actions/create-pull-request

Reply:
Well, the readme update with the new tag, we would want it in main?

Reply:
Looking at the README today, our install instructions reference the release by explicit version number, and it is published against `main`.